### DISCUSSION TITLE
Use heapless instead of a global allocator

### DISCUSSION THREAD

**Discussion by markszente** [slint-ui/slint#4234] [Ideas]
Hi,

If I'm correct, the `alloc` crate is presently used when you implement `Platform`. It would be cool to use Slint without having to use a global allocator. Fortunately, there is a crate for using such data structures with static memory allocation, called [`heapless`](https://docs.rs/heapless/latest/heapless/). Would it be possible to implement something like this as a feature flag?

**Comment by ogoffart** [comment]
Right now Slint uses a global allocator, and it would be a major undertaking to change that.
What is your use case for which you can't have a global allocator?
